Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -1017,6 +1017,19 @@ public final class RegistryClient: AsyncCancellable {
         timeout: DispatchTimeInterval? = .none,
         observabilityScope: ObservabilityScope
     ) async throws -> Set<PackageIdentity> {
+        // Validate that the source control URL is well-formed before making the request.
+        // This catches malformed URLs that may contain error messages (e.g., from git credential helpers).
+        // Git credential errors often produce messages like:
+        // "'https://github.com/owner/repo.git': failed looking up identity for ..."
+        // which can get concatenated with the URL.
+        guard let url = scmURL.url,
```

Comment:
wouldn't a better fix be to catch the errors when forming the `scmURL` over handling it here?